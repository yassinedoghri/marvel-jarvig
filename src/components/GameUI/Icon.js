import { pulse } from "react-animations";
import styled, { keyframes } from "styled-components";

const pulseAnimation = keyframes`${pulse}`;

const Icon = styled.span`
  color: ${props => props.theme.colors.contrastDark.primary};
  padding-left: 0.6em;
  padding-right: 0.3em;
  font-size: 1.8rem;

  & svg {
    animation: ${props =>
      props.animated ? `1s ${pulseAnimation} infinite` : "none"};
  }
`;

export default Icon;
