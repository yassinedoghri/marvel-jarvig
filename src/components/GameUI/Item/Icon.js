import styled from "styled-components";


const Icon = styled.span`
    color: ${props => props.theme.colors.dark.primary};
    padding: 0 .6em;
    font-size: 1.8rem;
`;

export default Icon;
