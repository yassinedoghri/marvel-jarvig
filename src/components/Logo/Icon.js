import styled from "styled-components";

const Icon = styled.img`
  height: 2.5em;
`;

export default Icon;
