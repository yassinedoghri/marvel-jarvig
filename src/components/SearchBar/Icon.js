import styled from "styled-components";


const Icon = styled.span`
    color: ${props => props.theme.colors.primary};
    padding: .6em;
    border-radius: 25px 0 0 25px;
`;

export default Icon;
