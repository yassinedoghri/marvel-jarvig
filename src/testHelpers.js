import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
