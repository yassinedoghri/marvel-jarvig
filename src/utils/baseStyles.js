import { normalize } from "polished";
import { injectGlobal } from "styled-components";

export default () => injectGlobal`
    ${normalize()}

    @import url("https://fonts.googleapis.com/css?family=Bangers");
    @import url("https://fonts.googleapis.com/css?family=Roboto:400,900");
    @import url("https://fonts.googleapis.com/css?family=Press+Start+2P");
  
    * {
      font-family: "Roboto", sans-serif;
      box-sizing: border-box;
    }
    
    *:focus {
      outline: none;
    }
    
    a {
      text-decoration: none;
    }
    
    ul li, ol li {
        list-style: none;
    }
    
    html, body, #root {
      height: 100%;
    }
    
    body {
      padding: 0;
      background-color: white;
    }
`;
