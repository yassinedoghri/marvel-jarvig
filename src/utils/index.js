export { default as style } from "utils/style";
export { default as baseStyles } from "utils/baseStyles";
export { default as helpers } from "utils/helpers";
export { default as testHelpers } from "utils/testHelpers";
export { default as registerServiceWorker } from "utils/registerServiceWorker";
export { default as media } from "utils/media";
