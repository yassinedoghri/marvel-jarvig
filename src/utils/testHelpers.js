import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import {
  ConnectedRouter,
  routerMiddleware,
  routerReducer
} from "react-router-redux";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import rootReducer from "reducers";
import { applyMiddleware, combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "theme";

const history = createMemoryHistory("/");
const routeMiddleware = routerMiddleware(history);

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  applyMiddleware(routeMiddleware)
);

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export function renderInContext(component) {
  return renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}

export function ShallowRenderInContext(component) {
  const renderer = new ShallowRenderer();
  return renderer.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}
