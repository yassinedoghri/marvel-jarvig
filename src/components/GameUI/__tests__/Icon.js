import Icon from "components/GameUI/Icon";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

describe("GameUI Icon", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Icon>Icon</Icon>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("adds animated modifier", () => {
    const tree = renderWithTheme(<Icon animated>Icon</Icon>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
