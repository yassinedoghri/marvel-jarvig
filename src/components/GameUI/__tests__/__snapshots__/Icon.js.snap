// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameUI Icon adds animated modifier 1`] = `
<span
  className="sc-bdVaJa cwMKVC"
>
  Icon
</span>
`;

exports[`GameUI Icon renders correctly 1`] = `
<span
  className="sc-bdVaJa iVzWHn"
>
  Icon
</span>
`;
